### PR TITLE
Get rid of janky error reraising when trying to print it's type

### PR DESCRIPTION
**Pull request by Tangdongle**
Fix for https://github.com/inim-repl/INim/issues/92

When printing the type for the line `proc adderNoReturnNoType(a: float, b: float) = a + b`, with showTypes enabled INim would try to print the type of the previous statement as something like `echo $(proc adderNoReturnNoType(a: float, b: float) = a + b) == $(type(proc adderNoReturnNoType(a: float, b: float) = a + b))`. 

Once this error is passed through proc showErrors to be printed, INim attempts to compile this statement and in turn hits another error. This PR prevents nested errors caused by bad type printing statements to not be re-raised, so we can output the correct error to the user.

### CODE REVIEW

**Comment by 0atman** [comment]
apologies @Tangdongle, could you fix the conflicted test? Looks good to me

**Comment by Tangdongle** [comment]
@0atman Fixed on this

File: tests/test_interface.nim [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## TODO: Split these up
 ## Maybe see if I can store a base state of the process before each tests runs and roll back after
-import osproc, streams, os
+import osproc, streams, os, sequtils
```
Comment (0atman) [inline tests/test_interface.nim:3]: `sequtils` isn't used, did you intend to?

**Comment by 0atman** [comment]
other than the sequtils warning, this looks good, thanks! :ok_hand:

File: tests/test_interface.nim (hunk 1 above)
Comment (Tangdongle) [reply tests/test_interface.nim]: I thought it might be required for `in`, but I guess not. Removed now

Comment (0atman) [reply tests/test_interface.nim]: nice, I saw the warning when running the tests :ok_hand:

**Comment by 0atman** [comment]
So this is a modest improvement, but I can't get the right type for your example, am I missing something?

This branch:
```nim
nim> proc adderNoReturnNoType(a: float, b: float) = a + b
Error: expression 'a + b' is of type 'float' and has to be discarded
```

Master:
```nim
nim> proc adderNoReturnNoType(a: float, b: float) = a + b
Error: expected: ')', but got: 'adderNoReturnNoType'
```

**Comment by 0atman** [comment]
Wait, now the tests are failing. O.o

**Comment by Tangdongle** [comment]
Hmmm, maybe I missed something when resolving conflicts

**Comment by Tangdongle** [comment]
`Error: expression 'a + b' is of type 'float' and has to be discarded` is the correct response. The `Error: expected: ')', but got: 'adderNoReturnNoType'` is from the line not compiling correctly with showTypes enabled. The "has to be discarded" is what the user wants to see

**Comment by 0atman** [comment]
OH! Then we are good there. Thanks for looking at the tests. Another win for CI! :ok_hand:

**Comment by Tangdongle** [comment]
@0atman Can we run the CI again? I can't reproduce the test failure on my machine.

**Comment by 0atman** [comment]
re-running

**Comment by 0atman** [comment]
hmm

```
[Suite] Interface Tests
Traceback (most recent call last)
/nim/lib/pure/unittest.nim(647) test_interface
/__w/INim/INim/tests/test_interface.nim(13) getResponse
/nim/lib/pure/streams.nim(364) writeLine
/nim/lib/pure/streams.nim(339) write
/nim/lib/pure/streams.nim(309) writeData
/nim/lib/pure/streams.nim(1219) fsWriteData
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
Error: execution of an external program failed: '/__w/INim/INim/tests/test_interface '
stack trace: (most recent call last)
/tmp/nimblecache/nimscriptapi.nim(168, 16)
/__w/INim/INim/inim_401.nims(26, 8) testTask
/nim/lib/system/nimscript.nim(260, 7) exec
/nim/lib/system/nimscript.nim(260, 7) Error: unhandled exception: FAILED: nim c -r -d:prompt_no_history tests/test_interface.nim [OSError]
     Error: Exception raised during nimble script execution
##[error]Process completed with exit code 1.
```

**Comment by 0atman** [comment]
Very odd, I can't replicate the error on my machine either.

**Comment by 0atman** [comment]
AHA! If I merge master into this branch then run the tests, I DO get the error:

```
[Suite] Interface Tests
Traceback (most recent call last)
/home/oatman/.choosenim/toolchains/nim-1.2.0/lib/pure/unittest.nim(647) test_interface
/home/oatman/projects/INim/tests/test_interface.nim(13) getResponse
/home/oatman/.choosenim/toolchains/nim-1.2.0/lib/pure/streams.nim(364) writeLine
/home/oatman/.choosenim/toolchains/nim-1.2.0/lib/pure/streams.nim(339) write
/home/oatman/.choosenim/toolchains/nim-1.2.0/lib/pure/streams.nim(309) writeData
/home/oatman/.choosenim/toolchains/nim-1.2.0/lib/pure/streams.nim(1219) fsWriteData
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
Error: execution of an external program failed: '/home/oatman/projects/INim/tests/test_interface '
stack trace: (most recent call last)
/tmp/nimblecache/nimscriptapi.nim(165, 16)
/home/oatman/projects/INim/inim_62159.nims(26, 8) testTask
/home/oatman/.choosenim/toolchains/nim-1.2.0/lib/system/nimscript.nim(260, 7) exec
/home/oatman/.choosenim/toolchains/nim-1.2.0/lib/system/nimscript.nim(260, 7) Error: unhandled exception: FAILED: nim c -r -d:prompt_no_history tests/test_interface.nim [OSError]
     Error: Exception raised during nimble script execution
```

**Comment by 0atman** [comment]
Can you replicate it this way?

**Comment by Tangdongle** [comment]
Can't believe I forgot how git works. I see it now, I'll see what's going on

**Comment by 0atman** [comment]
hum, still failing, what could the problem be?

**Comment by Tangdongle** [comment]
Whaaat ? Now it passes once I add logging (╯°□°)╯︵ ┻━┻? It should have passed last time, haha!

**Comment by 0atman** [comment]
Do you think this is ok to merge? Too spooky? :ghost:

**Comment by Tangdongle** [comment]
Well, I removed the logging and it's still passing. I think it should be OK

**Comment by 0atman** [comment]
I don't know what the problem was, tests work locally for me, lets see what happens on merge! :crossed_fingers:

**Comment by 0atman** [comment]
I THINK this is working on master?

```
λ git clone git@github.com:inim-repl/INim.git
Cloning into 'INim'...
remote: Enumerating objects: 116, done.
remote: Counting objects: 100% (116/116), done.
remote: Compressing objects: 100% (68/68), done.
remote: Total 691 (delta 49), reused 65 (delta 20), pack-reused 575
Receiving objects: 100% (691/691), 2.62 MiB | 5.23 MiB/s, done.
Resolving deltas: 100% (317/317), done.
~/projects via  v3.8.2 took 2s 

λ cd INim/
INim on  master via 👑 v1.2.0 

λ nimble build
  Verifying dependencies for inim@0.5.0
      Info: Dependency on cligen@>= 1.0.0 already satisfied
  Verifying dependencies for cligen@1.0.0
      Info: Dependency on noise@any version already satisfied
  Verifying dependencies for noise@0.1.3
   Building inim/inim using c backend
INim on  master [?] via 👑 v1.2.0 

λ ./inim --showTypes
👑 INim 0.5.0
Nim Compiler Version 1.2.0 [Linux: amd64] at /home/oatman/.nimble/bin/nim
nim> proc adderNoReturnNoType(a: float, b: float) = a + b
Error: expression 'a + b' is of type 'float' and has to be discarded
nim> 

Quitting INim: Goodbye!
```

**Comment by Tangdongle** [comment]
> I don't know what the problem was, tests work locally for me, lets see what happens on merge! crossed_fingers

Just for clarity's sake, the tests were failing because of a bad merge I performed (doh!). I was closing the input stream in the tests early (see https://github.com/inim-repl/INim/pull/96/commits/56a237183739e68ee9fa7ce8f9373d2898a6aaa5). Sorry about the roller-coaster :roller_coaster:

**Comment by 0atman** [comment]
Aha! Thanks for the explanation :grin:

**Pull request merged**